packages/components/src/components/error-modal/error-modal.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import DesktopWrapper from '../desktop-wrapper';
+import MobileWrapper from '../mobile-wrapper';
+import Modal from '../modal';
+import MobileDialog from '../mobile-dialog';
+import ErrorModalContent from './error-modal-content';
+
+type TMessageObject = {
+    message: string;
+    toString: () => string;
+};
+
+type TErrorModalProps = {
+    messages: Array<TMessageObject | React.ReactNode>;
+};
+
+const ErrorModal = ({ messages }: TErrorModalProps) => {
+    const [is_error_modal_open, setErrorModalOpen] = React.useState<boolean>(false);
```

Comment:
@hamza-deriv TS can infer the boolean type from state itself, no need to mention it

Reply:
sure